### ISSUE TITLE
copy-pasting a subflow instance with credentials properties causes flow to not start properly

### ISSUE THREAD

**Issue by Fadoli** [node-red/node-red#2783]
### What are the steps to reproduce?

1. Create a subflow template with a credential input
1. Use the template somewhere
1. Copy-paste the use of the template to duplicate it
1. Deploy

### What happens?

The credential input property seems to be left out (expected), but causes the start of the node to behave incorrectly

```
TypeError: Cannot read property 'hasOwnProperty' of undefined
    at Subflow.subflowInstance.env.forEach.e (/usr/src/node-red/node_modules/@node-red/runtime/lib/flows/Subflow.js:141:58)
    at Array.forEach (<anonymous>)
    at new Subflow (/usr/src/node-red/node_modules/@node-red/runtime/lib/flows/Subflow.js:133:38)
    at Object.createSubflow [as create] (/usr/src/node-red/node_modules/@node-red/runtime/lib/flows/Subflow.js:508:12)
    at Flow.start (/usr/src/node-red/node_modules/@node-red/runtime/lib/flows/Flow.js:205:55)
    at start (/usr/src/node-red/node_modules/@node-red/runtime/lib/flows/index.js:345:33)
    at stop.then.then (/usr/src/node-red/node_modules/@node-red/runtime/lib/flows/index.js:210:21)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

### What do you expect to happen?

The flow start should not fail, and the credential props should be considered as empty. Or the state of the configuration of the subflow should be invalid (because the credential prop is missing or something in the like). Or the error should instruct the user about setting the credentials props ?

As a side-note : opening the configuration and applying it fix the issue by creating an empty credential value

### Please tell us about your environment:

- [ ] Node-RED version: 1.2.6
- [ ] Node.js version: 10.23.0 (also tested on node 14, it shouldn't matter)
- [ ] npm version: doesn't matter
- [ ] Platform/OS: Linux and using DockerImage
- [ ] Browser: Firefox 83.0 (should not matter)

**Comment by aurelhann** [comment]
Same for me. 

Frustrating :(

**Comment by knolleary** [comment]
Thanks for reporting it. I've pushed the fix - will be in 1.2.7, although we don't have a planned release date for that yet.

**Comment by TechupBusiness** [comment]
Wow this bug is killing me... and then I discovered another user experience killer... when I click on "Delete Subflow" (because I needed to delete all copies of the subflow in my main flow and clicked once wrong), there is no confirmation if I wanna delete. Now my work of hours is gone.

**Comment by Fadoli** [comment]
@TechupBusiness FYI this is fixed in 1.2.7 which has been released
